app/utilities/resource_packager.rb

```diff
@@ -19,11 +19,17 @@ def activities
     @lesson_parts.values
   end
 
+  def slide_decks
+    activities.map(&:slide_deck)
+  end
+
   def slide_deck_tempfiles
-    activities.map { |activity| activity.slide_deck.attachment.open { |f| f } }
+    slide_decks.map { |activity| activity.attachment.open { |f| f } if activity.attached? }.compact
```

Comment:
This line feels a bit complicated to me, the cognitive load is too high.

Could breaking it up improve it? Something along the lines of this.

```suggestion
    slide_decks
      .select { |activity| activity.attached? }
      .map { |activity| activity.attachment.blob.download }
```

I'm not actually sure if `blob.download` is actually equivalent to `attachment.open { |f| f }` here but it's just an example.